src/thor/bidirectional_astar.cc

```diff
@@ -581,8 +569,7 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
         if (opp_status.set() == EdgeSet::kPermanent ||
             (opp_status.set() == EdgeSet::kTemporary &&
              edgelabels_reverse_[opp_status.index()].predecessor() == kInvalidLabel)) {
-          if (SetForwardConnection(graphreader, fwd_pred) &&
-              opp_status.set() == EdgeSet::kPermanent) {
+          if (SetForwardConnection(graphreader, fwd_pred)) {
```

Comment:
this seems like a bug to me. in the line above we're explicitly allowing `kTemporary` under the condition that the reverse tree is exhausted (which I think makes sense or at least made sense when I changed it 2 weeks ago), but then one line below we're again not allowing `kTemporary`. we should choose one way to do it, so this is just a best guess, I can also remove the 2-3 lines above instead

Reply:
with all the changes to this algorithm we need to also check it for regressions by running a befor and after route set.

i have messed with these lines too before and confused myself about what is the right thing

Reply:
Ok will do a check with a bigger request file and diff the narrative

Reply:
also check the perf diff (can be done at the same time with the same script by running with `time`)